db/migrate/20200428205535_uniqueness_rule_references_rule_index.rb

```diff
@@ -0,0 +1,28 @@
+class UniquenessRuleReferencesRuleIndex < ActiveRecord::Migration[5.2]
+  class NewRuleReferencesRule < ApplicationRecord; end
+
+  def change
+    ids_to_keep = RuleReferencesRule.group(
+      'rule_id, rule_reference_id'
+    ).having("count(*) > 1").pluck(Arel.sql('MIN(id)'))
+    create_table :new_rule_references_rules, id: false do |t|
+      t.references :rule, type: :uuid, index: true, null: false
+      t.references :rule_reference, type: :uuid, index: true, null: false
+    end
+    columns = [:rule_id, :rule_reference_id]
+    batch_size = 1000
+    to_keep = RuleReferencesRule.where(id: ids_to_keep)
+    batches_to_run = to_keep.count/batch_size
+    to_keep.in_batches(of: batch_size).each_with_index do |batch, index|
```

Comment:
`find_each` by default has a batch size of 1000. You can use `find_each.with_index`